Gruntfile.js

```diff
@@ -1591,10 +1589,10 @@ module.exports = function(grunt) {
 					encoding: 'utf8',
 				} );
 				// `data:` URLs are allowed:
-				const match = contents.match( /sourceMappingURL=((?!data:).)/ );
+				const doesNotHaveSourceMap = ! /^\/\/# sourceMappingURL=((?!data:).)/m.test(contents);
```

Comment:
A couple of important changes here.

First, a minor change. We use [`RegExp.test()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test) instead of [`String.match()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match). We don't care about the result of matching, we want to know "does this match," so `RegExp.test()` (with its boolean result) is ideal.

Second, and more important. The expression was not specific enough and was matching JavaScript source files that were, themselves, looking for sourcemap comments. In this PR is was matching this line from [es-module-shims](https://unpkg.com/browse/es-module-shims@1.8.2/dist/es-module-shims.wasm.js):

```js
  const sourceMapURLCommentPrefix = '\n//# sourceMappingURL=';
```

We change to better match a source map comment by anchoring the expression to the start of the line and matching the comment opening `//# `. This fixes the false positives and allows us to remove several files from the ignore list.